pkg/reconciler/revision/reconcile_resources.go

```diff
@@ -72,7 +76,10 @@ func (c *Reconciler) reconcileDeployment(ctx context.Context, rev *v1.Revision)
 		return fmt.Errorf("failed to update deployment %q: %w", deploymentName, err)
 	}
 
-	rev.Status.PropagateDeploymentStatus(&deployment.Status)
+	// When we are scaling down we want to keep the error that we might have seen before
+	if *deployment.Spec.Replicas > 0 {
```

Comment:
`!rev.Status.IsActivationRequired()` is not not enough because we go back to the replicaset failure issue and we want to move away from conditioning.